anndata/_core/anndata.py

```diff
@@ -1422,7 +1421,7 @@ def copy(self, filename: Optional[PathLike] = None) -> 'AnnData':
                 varm=self.varm.copy(),
                 obsp=self.obsp.copy(),
                 varp=self.varp.copy(),
-                raw=self.raw,  # copies automatically
+                raw=self.raw.copy(),
```

Comment:
I think this should be:

```python
raw=self.raw.copy() if self.raw is not None else None,
```

Reply:
Oh. We never test `.copy()`? I’ll add a test